sycl/doc/extensions/C-CXX-StandardLibrary/DeviceLibExtensions.rst

```diff
@@ -32,3 +32,159 @@ Arguments:
 Example of a message:
 .. code:
    foo.cpp:42: void foo(int): global id: [0,0,0], local id: [0,0,0] Assertion `buf[wiID] == 0 && "Invalid value"` failed.
+
+cl_intel_devicelib_math
```

Comment:
Not all functions from math.h are supported, specifically:
  - abs
  - ceil
  - copysign
  - fabs
  - floor
  - fmax
  - fmin
  - llrint
  - llround
  - lrint
  - lround
  - nan
  - nearbyint
  - nexttoward
  - rint
  - round
  - scalbln
  - scalbn
  - trunc

Reply:
Hi, Andrew. In the long term, we can support them. So, I think we can mention that support for those functions is  in future plan.

Reply:
Please list them in DeviceLibExtensions.rst as well.
These extensions are supposed to be a "contract" between the FE and a device compiler. We should not add more functions to a particular extension, otherwise we invalidate device compilers that support an older version.

Reply:
Hi, Andrew. So I need to claim that:
1. we don't support all functions in math.h in cl_intel_devicelib_math extension
2. List the functions which we plan to support in the future
in DeviceLibExtensions.rst?
Can I add a "Note:" section in cl_intel_devicelib_math to claim those 2 points? Like following:
Arguments:
Those __devicelib_* functions have the same argument type and return type as corresponding
math functions from <math.h> or <cmath>.

Note:
Currently, not all functions from <math.h>; or <cmath> are supported in this extension. Following functions will be supported in the future:
.. code:
   int    __devicelib_abs(int x);
   float  __devicelib_ceilf(float x);
   double __devicelib_ceil(double x);
   float  __devicelib_copysign(float x, float y);
   double __devicelib_copysign(double x, double y);
   float  __devicelib_fabsf(float x);
   double __devicelib_fabs(double x);
   float  __devicelib_floorf(float x);
   double __devicelib_floor(double x);
   float  __devicelib_fmaxf(float x);
   double __devicelib_fmax(double x);
   float  __devicelib_fminf(float x);
   double __devicelib_fmin(double x);
   float  __devicelib_nextafterf(float x, float y);
   double __devicelib_nextafter(double x, double y);
   float  __devicelib_rintf(float x);
   double __devicelib_rint(double x);
   float  __devicelib_roundf(float x);
   double __devicelib_round(double x);
   float  __devicelib_truncf(float x);
   double __devicelib_trunc(double x);

Thank you very much.

Reply:
> Note:
> Currently, not all functions from <math.h>; or are supported in this extension. Following functions will be supported in the future:
> .. code:
> int __devicelib_abs(int x);
> float __devicelib_ceilf(float x);
> double __devicelib_ceil(double x);
> [...]

Sounds good.

Reply:
Just added.
Thank you very much.